front-end/src/screens/CreatePost/index.tsx

```diff
@@ -23,20 +26,36 @@ interface Props {
 	className?: string
 }
 
+const TWO_WEEKS = 2 * 7 * 24 * 60 * 60 * 1000;
+
 const CreatePost = ({ className }:Props): JSX.Element => {
 	const [title, setTitle] = useState('');
 	const [content, setContent] = useState('');
-	// const [hasPoll, setHasPoll] = useState(false);
+	const [hasPoll, setHasPoll] = useState(false);
 	const { queueNotification } = useContext(NotificationContext);
 	const [selectedTopic, setSetlectedTopic] = useState(1);
 	const currentUser = useContext(UserDetailsContext);
 	const { control, errors, handleSubmit } = useForm();
+	const network = getNetwork();
+	const { api, isApiReady } = useContext(ApiPromiseContext);
+	const DEFAULT_TIME = chainProperties?.[network]?.blockTime;
+	const [blocktime, setBlocktime] = useState(DEFAULT_TIME);
 
+	const { data } = useGetCurrentBlockNumberQuery();
 	const [createPostMutation, { loading, error }] = useCreatePostMutation();
 	const [postSubscribeMutation] = usePostSubscribeMutation();
 	const [isSending, setIsSending] = useState(false);
 	const { history } = useRouter();
 
+	useEffect(() => {
+
+		if (!isApiReady) {
+			return;
+		}
+
+		setBlocktime(api.consts.babe?.expectedBlockTime.toNumber());
+	}, [api, isApiReady]);
+
```

Comment:
This is the same code than `/src/components/BlockCountdown.tsx` (and `src/components/BlocksToTime.tsx`).
Can we have a `getBlockTime` helper function in `utils` to replace those duplications?

Reply:
Would need context for this as api.consts.babe?.expectedBlockTime can be called only in useEffect.

Reply:
I guess you can create a hook `useBlockTime`. Also the same way `src/hooks/useRouter.ts`, with a `useMemo` so that we don't make the call all the time if the api hasn't changed.

Reply:
Done